cql3/selection/selection.cc

```diff
@@ -297,19 +297,38 @@ class selection_with_processing : public selection {
     }
 
     virtual std::vector<shared_ptr<functions::function>> used_functions() const override {
-        return selectors_with_processing(_factories).used_functions();
+        auto ret = std::vector<shared_ptr<functions::function>>();
+        expr::recurse_until(expr::tuple_constructor{_selectors}, [&] (const expr::expression& e) {
+            if (auto fc = expr::as_if<expr::function_call>(&e)) {
+                auto func = std::get<shared_ptr<functions::function>>(fc->func);
+                ret.push_back(func);
+                if (auto agg_func = dynamic_pointer_cast<functions::aggregate_function>(std::move(func))) {
+                    auto& agg = agg_func->get_aggregate();
+                    if (agg.aggregation_function) {
+                        ret.push_back(agg.aggregation_function);
+                    }
+                    if (agg.state_to_result_function) {
+                        ret.push_back(agg.state_to_result_function);
```

Comment:
What about the state_reduction_function - don't you need it too?

Reply:
No, it's never used when evaluating a query. This is also what selectors_with_processing::used_functions() returns.

The reductions are never prepared, always materialized at the point of evaluation. There's a case to do it differently, but I don't want to mix that into the series.